uriget/uriget.go

```diff
@@ -244,25 +244,45 @@ func (o *options) getGit(ctx context.Context, u *url.URL) ([]byte, error) {
 func (o *options) getOci(ctx context.Context, u *url.URL) ([]byte, error) {
 	ref, err := registry.ParseReference(u.Host + u.Path)
 	if err != nil {
-		return nil, fmt.Errorf("can't parse artifact URL into a valid reference: %w", err)
+		return nil, fmt.Errorf("invalid artifact URL: %w", err)
 	}
-	store, err := oci.New(o.tempDir)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create OCI layout store: %w", err)
+	if ref.Reference == "" {
+		ref.Reference = "latest"
 	}
+	specifiedFile := strings.TrimPrefix(u.Fragment, "#")
 	remoteRepo, err := remote.NewRepository(ref.String())
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to remote repository: %w", err)
+		return nil, fmt.Errorf("connection to remote repository failed: %w", err)
+	}
+	_, rc, err := remoteRepo.Manifests().FetchReference(ctx, ref.Reference)
+	if err != nil {
+		return nil, fmt.Errorf("manifest fetch failed: %w", err)
 	}
-	tag := "latest"
-	if ref.Reference != "" {
-		tag = ref.Reference
+	defer rc.Close()
+	var manifest v1.Manifest
+	if err = json.NewDecoder(rc).Decode(&manifest); err != nil {
+		return nil, fmt.Errorf("manifest decode failed: %w", err)
+	}
+	var selectedLayer *v1.Descriptor
+	for _, layer := range manifest.Layers {
+		title := layer.Annotations[v1.AnnotationTitle]
+		if (specifiedFile != "" && title == specifiedFile) ||
+			(specifiedFile == "" && strings.HasSuffix(title, ".yaml")) {
+			selectedLayer = &layer
```

Comment:
If `specifiedFile` is blank, and `selectedLayer` is already set, I think we should throw an error indicating that the user should pick a specific file from within the manifest.

For example: `manifest contains more than 1 yaml file, a specific file reference must be set by the url fragment`